### ISSUE TITLE
seating arrangement 

### ISSUE THREAD

**Issue by josephniranjan** [Tomaszal/HackerEarth#3]
may i know how you derive this formula below for Hackerearth I/O "seating arrangement" exercise?
n = n + 2 * (6 - (n - 1) % 12) - 1

**Comment by Tomaszal** [comment]
I might be a little late and I've done this problem years ago so I don't quite remember how I came to the conclusion, but I'll try to explain the formula the best I can.

First of all, we narrow down the seat position relative to it's row. So for example seat **23** is in the same position relative to it's row as 35 and 11. That's simple to do, since there is 12 seats in a row we just take a modulus of the seat number. 23 % 12 = 11. However, if we do this with a position that is divisible by 12 (f.e. 24), we will get 0, which is not what we want (we want 12). This is easily fixable by subtracting 1 before taking the modulus. (24 - 1) % 12 = 11. Now a position relative to it's side starts from 0 instead of 1 (0-11 instead of 1-12). There's 6 seats on one side, so to find a seat position relative to it's side on the opposite side we just subtract the position relative to it's row from 6. That'll give us a number in range of -5 to 6, depending on the position's side. 6 - (**23** - 1) % 12 = -4. Since there's 2 sides on a row we have to multiply that result by 2. -4 * 2 = -8. Now we got a seat position relative to our initial seat position that should be on the opposite side. But since we subtracted 1 before taking the modulus we introduced a little mistake, which is easily fixed by subtracting 1 again. 23 + (-8) - 1 = 14.

Hope that helps.
